CapstoneProject/Assets/Scripts/Core/Entity/Objects/Item.cs

```diff
@@ -17,19 +21,18 @@ private void Start()
             animator = GetComponentInChildren<Animator>();
             haveBeenInteracted = false;
             animator.SetBool("Idle", true);
+            _player = FindObjectOfType<Player>();
         }
 
         public void Interact()
         {
-            if (haveBeenInteracted) return;
+            if (haveBeenInteracted || _player.GetComponent<KeyItemComponent>().TryToUseKeys() == false) return;
             animator.SetBool("Idle", false);
             animator.SetTrigger("Open");
             haveBeenInteracted = true;
-        }
-
-        private void OnTriggerStay2D(Collider2D other)
-        {
-            
+            _player.GetComponent<KeyItemComponent>().TryToUseKeys();
```

Comment:
try use keys returns a bool. if they don't have a key right now you still give them the coins and open? but if they have a key you take it away?

Reply:
I see you do this above in your check though, so if they have 2 keys, you'd use both on the one chest :(